src/ai/backend/agent/kernel.py

```diff
@@ -926,49 +940,62 @@ async def read_output(self) -> None:
             try:
                 msg_type, msg_data = await self.output_sock.recv_multipart()
                 try:
-                    if msg_type == b"status":
-                        await self.status_queue.put(msg_data)
-                    elif msg_type == b"completion":
-                        await self.completion_queue.put(msg_data)
-                    elif msg_type == b"service-result":
-                        await self.service_queue.put(msg_data)
-                    elif msg_type == b"model-service-result":
-                        await self.model_service_queue.put(msg_data)
-                    elif msg_type == b"apps-result":
-                        await self.service_apps_info_queue.put(msg_data)
-                    elif msg_type == b"stdout":
-                        if self.output_queue is None:
-                            continue
-                        if len(msg_data) > self.max_record_size:
-                            msg_data = msg_data[: self.max_record_size]
-                        await self.output_queue.put(
-                            ResultRecord(
-                                "stdout",
-                                decoders[0].decode(msg_data),
+                    match msg_type:
+                        case b"status":
+                            await self.status_queue.put(msg_data)
+                        case b"completion":
+                            await self.completion_queue.put(msg_data)
+                        case b"service-result":
+                            await self.service_queue.put(msg_data)
+                        case b"model-service-result":
+                            await self.model_service_queue.put(msg_data)
+                        case b"model-service-status":
+                            response = json.loads(msg_data)
+                            event: AbstractEvent
+                            if response["is_healthy"]:
+                                event = KernelHealthyEvent(
+                                    self.kernel_id, self.session_id, reason=response["model_name"]
+                                )
+                            else:
+                                event = KernelHealthCheckFailedEvent(
+                                    self.kernel_id, self.session_id, reason=response["model_name"]
+                                )
```

Comment:
IMO, it would be better to unify these two events and distinguish the status by a field of the event to avoid bloating the kind of events.

Reply:
Q. Do you have plans to expand the usage of "kernel health" events in other contexts?

Reply:
I've updated the implementation to both unify the event type and create a new event argument solely for representing model service health status.